design/api-design.md

```diff
@@ -0,0 +1,47 @@
+# API design
+
+## Requirements
+
+* API should be relatively easy to call (i.e., through a single GET/POST request)
+* It should be reasonably responsive by default (<5s)
+* API returns JSON object
+* If the schema is expected to change a lot, versioned API routes would be preferable.
+
+## Querying the API
+
+API calls are contained to one call per query. Front-end calls could look like `api.epiverse.org/query="how to estimate outbreak probability of dengue in colombia in next three months"&filter="epiverse"` (versioned could look like `api.epiverse.org/v1/query[...]`).
+
+## Response
+
+```json
+{
+  "query": "how to estimate outbreak probability of dengue in colombia in next three months",
+  "filter": "epiverse",
+  "response": {
+    "text": "In order to estimate the outbreak probability of dengue in Colombia in the next three months, you can use the following Epiverse packages",
+    "results": [
+      {
+        "package": "epidemics",
+        "logo": "https://epiverse-trace.github.io/epidemics/logo.svg",
+        "website": "https://epiverse-trace.github.io/epidemics",
+        "source": "https://github.com/epiverse-trace/epidemics",
+        "vignettes": [
+            "https://epiverse-trace.github.io/epidemics/articles/modelling_interventions.html",
+            "https://epiverse-trace.github.io/epidemics/articles/modelling_multiple_interventions.html"
+        ],
```

Comment:
Getting slightly out of scope for this PR but if this information is returned directly from the backend, the scraper should collect it.

This can be turned into a separate issue along the lines of "Metadata scraper" (in contrast with "Documentation scraper")